src/Groups/matrices/linear_centralizer.jl

```diff
@@ -0,0 +1,385 @@
+
+########################################################################
+#
+# Orders
+#
+########################################################################
+
+function _GL_order(n::Int, q::fmpz)
+   res = q^div(n*(n-1),2)
+   for i in 1:n res *= (q^i-1) end
+   return res
+end
+
+_GL_order(n::Int, F::Ring) = _GL_order(n, order(F))
+
+function _SL_order(n::Int, q::fmpz)
+   res = q^div(n*(n-1),2)
+   for i in 2:n res *= (q^i-1) end
+   return res
+end
+
+_SL_order(n::Int, F::Ring) = _SL_order(n, order(F))
+
+########################################################################
+#
+# Centralizer in GL
+#
+########################################################################
+
+# Compute a generating set for every element in GL(n,q) and SL(n,q)
+# generators for GL(n,q) and SL(n,q) are described in:  Taylor, D. E., Pairs of Generators for Matrix Groups. I, The Cayley Bulletin, 3 (1987)
+
+# returns as matrices
+function _gens_for_GL(n::Int, F::Ring)
+   n !=1 || return [matrix(F,1,1,[primitive_root(F)])]
+   h1 = identity_matrix(F,n)
+   h2 = zero_matrix(F,n,n)
+   if order(F)==2
+      h1[1,2] = 1
+   else
+      h1[1,1] = primitive_root(F)
+      h2[1,1] = -1
+   end
+   h2[1,n] = 1
+   for i in 1:n-1 h2[i+1,i] = -1 end
+   return h1,h2
+end
+
+# returns as matrices
+# does the same as above with F = F[x]/(f), but every entry is replaced by a diagonal join of D corresponding blocks
+# ASSUMPTION: deg(f) > 1
+function _gens_for_GL_matrix(f::PolyElem, n::Int, F::Ring; D=1)
+   C = companion_matrix(f)
+   CP = evaluate(_centralizer(f),C)            # matrix of maximal order in the centralizer of the companion matrix
+
+   if n==1 return [diagonal_join([CP for i in 1:D])] end
+   h1 = identity_matrix(F,n*degree(f)*D)
+   insert_block!(h1,diagonal_join([CP for i in 1:D]),1,1)
+   h2 = zero_matrix(F,n*degree(f)*D,n*degree(f)*D)
+   for i in 1:(n-1)*degree(f)*D h2[i+degree(f)*D,i]=-1 end
+   for i in 1:degree(f)*D
+      h2[i,i]=-1
+      h2[i,i+(n-1)*degree(f)*D]=1
+   end
+   return h1,h2
+end
+
+# generators for centralizer of unipotent elements are described in: Giovanni De Franceschi, Centralizers and conjugacy classes in finite classical groups, arXiv:2008.12651
+
+# V = vector of integers of the dimensions of Jordan blocks
+# return the generators for the centralizers of the diagonal join of unipotent Jordan blocks of dimensions V
+# assumes V is sorted (e.g. [1,1,1,2,3,3])
+function _centr_unipotent(F::Ring, V::AbstractVector{Int}; isSL=false)
```

Comment:
So based on what you write elsewhere, perhaps this change should be made:
```suggestion
function _centr_unipotent(F::FinField, V::AbstractVector{Int}; isSL=false)
```
And similarly elsewhere?